File: Array and String/Add_Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        def sumBinary(*bin_digits):
            count_1 = 0
            count_0 = 0
            for x in bin_digits:
                if x == "1":
                    count_1 += 1
                    
            if count_1 == 0:
                ans = carry = 0
            elif count_1 == 1:
                ans = 1
                carry = 0
            elif count_1 == 2:
                ans = 0
                carry = 1
            elif count_1 == 3:
                ans = 1
                carry = 1
            
            return str(ans), str(carry)
        
        a_list = list(a)
        b_list = list(b)
        
        la = len(a_list)
        lb = len(b_list)
        
        if la>lb:
            b_list = ["0"]*(la-lb) + b_list
        elif lb>la:
            a_list = ["0"]*(lb-la) + a_list

        
        ans = []
        carry = "0"
        while a_list and b_list:
            bit_ans, carry = sumBinary(a_list.pop(), b_list.pop(), carry)
            ans.insert(0, bit_ans)
        
        if carry == "1":
            ans.insert(0, carry)
        return "".join(ans)

File: Array and String/Add_Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string s = "";
        for(int i=a.size()-1, j=b.size()-1, carry=0; i>=0 || j>=0 || carry; i--, j--, carry/=10) {
            int ni = i>=0?a[i]-'0':0;
            int nj = j>=0?b[j]-'0':0;
            int r = 00;
            if(ni&&nj)
                r=10;
            else if(ni^nj)
                r=01;
            if(carry) {
                if(r==01)
                    r=10;
                else
                    r+=01;
            }
            s = to_string(r%10) + s;
            carry = r;
        }
        return s;
    }
};